dev-games/tiled/tiled-1.11.0.ebuild

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..13} )
+inherit toolchain-funcs multiprocessing python-single-r1 xdg
+
+DESCRIPTION="A general purpose tile map editor"
+HOMEPAGE="https://www.mapeditor.org/"
+SRC_URI="https://github.com/mapeditor/tiled/archive/v${PV}/${P}.tar.gz"
+
+LICENSE="BSD BSD-2 GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="python examples"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="
+	app-arch/zstd
+	dev-qt/qtbase:6[X]
+	dev-qt/qtdeclarative:6
+	dev-qt/qtsvg:6
+	sys-libs/zlib
+	python? ( ${PYTHON_DEPS} )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-util/qbs
+"
+
+pkg_setup() {
+	if use python; then
+		python-single-r1_pkg_setup
+	fi
+}
+
+src_configure() {
+	if use python; then
+		eapply "${FILESDIR}"/${P}-python.patch
+	fi
+
+	qbs setup-qt /usr/bin/qmake6 qt6 || die
+	qbs config defaultProfile qt6 || die
+
+	if $(tc-is-clang); then
+		qbs setup-toolchains clang clang || die
+		qbs config profiles.qt6.baseProfile clang || die
+	else
+		qbs setup-toolchains gcc gcc || die
+		qbs config profiles.qt6.baseProfile gcc || die
+	fi
+}
+
+qbs_format_flags() {
+	local result="["
+
+	for flag in $@; do
+		if [[ ${result} != "[" ]]; then
+			result="${result},"
+		fi
+		result="${result}\"${flag}\""
+	done
+
+	result="${result}]"
+	echo ${result}
+}
+
+src_compile() {
+	set -o noglob
+	local QBS_CFLAGS=$(qbs_format_flags ${CFLAGS})
+	local QBS_CXXFLAGS=$(qbs_format_flags ${CXXFLAGS})
+	set +o noglob
```

Comment:
We could use meson-format-array for this?

https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/meson.eclass#n87

```
$ meson-format-array $(portageq envvar CFLAGS)
['-O3', '-march=znver2', '-pipe', '-flto=auto', '-fuse-linker-plugin', '-frecord-gcc-switches', '-Werror=odr', '-Werror=lto-type-mismatch', '-Werror=strict-aliasing']
```

Reply:
```
['-march=znver4', '-O3', '-pipe', '-flto=thin', '-Werror=odr', '-Werror=strict-aliasing']
ERROR: Unexpected command line parameter ''-O3','.
Expected an assignment of the form <property>:<value>, profile:<profile-name> or config:<configuration-name>.
```
Looks like it doesn't work with this format

Reply:
```diff
diff --git a/dev-games/tiled/tiled-1.11.0.ebuild b/dev-games/tiled/tiled-1.11.0.ebuild
index 879d2ab02c39..264ae23afc17 100644
--- a/dev-games/tiled/tiled-1.11.0.ebuild
+++ b/dev-games/tiled/tiled-1.11.0.ebuild
@@ -53,32 +53,24 @@ src_configure() {
 }
 
 qbs_format_flags() {
-       local result="["
+       local -a array
 
-       for flag in $@; do
-               if [[ ${result} != "[" ]]; then
-                       result="${result},"
-               fi
-               result="${result}\"${flag}\""
+       for flag in ${@}; do
+       array+=( "\"${flag}\"" )
        done
 
-       result="${result}]"
-       echo ${result}
+
+       echo "[$(IFS=","; echo "${array[*]}")]"
 }
 
 src_compile() {
-       set -o noglob
-       local QBS_CFLAGS=$(qbs_format_flags ${CFLAGS})
-       local QBS_CXXFLAGS=$(qbs_format_flags ${CXXFLAGS})
-       set +o noglob
-
        qbs build \
                qbs.installPrefix:"/usr" \
                projects.Tiled.useRPaths:false \
                projects.Tiled.installHeaders:true \
                project.libDir:$(get_libdir) \
-               modules.cpp.cFlags:${QBS_CFLAGS} \
-               modules.cpp.cxxFlags:${QBS_CXXFLAGS} \
+               modules.cpp.cFlags:"$(qbs_format_flags ${CFLAGS})" \
+               modules.cpp.cxxFlags:"$(qbs_format_flags ${CFLAGS})" \
                -j $(get_makeopts_jobs) \
                || die
 }
```

What would you think about this? (Working on a older version of the PR, will update in a bit).